megamek/src/megamek/client/bot/princess/Princess.java

```diff
@@ -610,7 +610,8 @@ protected void calculateFiringTurn() {
                     // damageMap for the target enemy.
                     for(WeaponFireInfo shot : plan) {
                         int targetId = shot.getTarget().getTargetId();
-                        double newDamage = damageMap.get(targetId) + shot.getExpectedDamage();
+                        double existingTargetDamage = damageMap.containsKey(targetId) ? damageMap.get(targetId) : 0;
```

Comment:
If `damageMap` implements `Map<K,V>` you could use [`getOrDefault`](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#getOrDefault-java.lang.Object-V-):
```suggestion
                        double existingTargetDamage = damageMap.getOrDefault(targetId, 0.0);
```

Reply:
It required an extra line of code change, but done.